server/controllers/content/content.ts

```diff
@@ -415,16 +422,15 @@ export default class Content {
             for (let content of contents) {
                 listOfContentIds.push(content.identifier);
             }
-            let filters = { identifier: listOfContentIds };
-            logger.debug(`ReqId = "${reqId}": Search all the contents in DB using content Id's`)
-            await this.searchInDB(filters, reqId)
+            logger.debug(`ReqId = "${reqId}": Search downloaded and downloading  contents in DB using content Id's`)
+            await this.searchDownloadingContent(listOfContentIds, reqId)
                 .then(data => {
-                    logger.info(`ReqId = "${reqId}": Found the ${data.docs.length} contents in ContentDb`)
+                    logger.info(`ReqId = "${reqId}": Found the ${data.docs.length} contents in Content_Download Db`)
                     for (let doc of data.docs) {
                         for (let content of contents) {
-                            logger.debug(`include addedToLibrary property for the contents which are downloaded`)
-                            this.includeAddedToLibraryProperty(doc, content, reqId);
-                            logger.info(`ReqId = "${reqId}": included addedToLibrary property for the contents which are downloaded`)
+                            if (doc.contentId === content.identifier) {
+                                content.downloadStatus = DOWNLOAD_STATUS[doc.status];
```

Comment:
don't send or have a hard dependency from backend status with the front end, have some abstraction.